lib/matplotlib/tests/test_axes.py

```diff
@@ -7574,6 +7575,26 @@ def test_empty_line_plots():
     assert len(line) == 1
 
 
+@pytest.mark.parametrize('err, fmt, match',
+                         ((ValueError, "foo",
+                           "Unrecognized character f in format string 'foo'"),
+                          (ValueError, "oo",
+                           "Illegal format string 'oo'; two marker symbols"),
+                          (ValueError, "::",
+                           "Illegal format string '::'; two linestyle "
+                           "symbols"),
+                          (ValueError, "rr",
+                           "Illegal format string 'rr'; two color symbols"),
+                          (ValueError, ":o:r",
+                           "Illegal format string ':o:r'; two linestyle "
+                           "symbols"),
+                          ))
```

Comment:
Suggested format:

```suggestion
@pytest.mark.parametrize('err, fmt, match', [
    (ValueError, "foo", "Unrecognized character f in format string 'foo'"),
    (ValueError, "oo", "Illegal format string 'oo'; two marker symbols"),
    (ValueError, "::", "Illegal format string '::'; two linestyle symbols"),
    (ValueError, "rr", "Illegal format string 'rr'; two color symbols"),
    (ValueError, ":o:r", "Illegal format string ':o:r'; two linestyle "
                         "symbols"),
])
```
You may also leave out `ValueError` because it's the same for all tests.

I suggest to add a test for different two markers as well, e.g. `o+`.

Reply:
Sorted out! (Kept it as a tuple for simplicity and performance reasons.)